shared/nm-utils/nm-test-utils.h

```diff
@@ -670,6 +670,20 @@ nmtst_test_quick (void)
 	} G_STMT_END
 #endif
 
+#define NMTST_EXPECT(domain, level, msg)        g_test_expect_message (domain, level, msg)
+
+#if (NETWORKMANAGER_COMPILATION) & NM_NETWORKMANAGER_COMPILATION_WITH_LIBNM_UTIL
+#define NMTST_EXPECT_LIBNM_U(level, msg)        NMTST_EXPECT ("libnm-util", level, msg)
+#define NMTST_EXPECT_LIBNM_G(level, msg)        NMTST_EXPECT ("libnm-glib", level, msg)
+
+#define NMTST_EXPECT_LIBNM_U_CRITICAL(msg)      NMTST_EXPECT_LIBNM_U (G_LOG_LEVEL_CRITICAL, msg)
+#define NMTST_EXPECT_LIBNM_G_CRITICAL(msg)      NMTST_EXPECT_LIBNM_G (G_LOG_LEVEL_CRITICAL, msg)
+#else
```

Comment:
No strong opinion, but I don't find those macros very useful; they are few characters shorter, but the name encodes the same information as the original function, with the disadvantage that you have to look up what they do if you don't know it.

Reply:
I mostly find the `NMTST_EXPECT_NM_*` macros useful, because they also check for the right logging level, so we get
```
-    g_test_expect_message ("NetworkManager", G_LOG_LEVEL_MESSAGE,
-                           "*<warn> * keyfile: 802-1x.client-cert: certificate or key file '/CASA/dcbw/Desktop/certinfra/client.pem' does not exist*");
+    NMTST_EXPECT_NM_WARN ("keyfile: 802-1x.client-cert: certificate or key file '/CASA/dcbw/Desktop/certinfra/client.pem' does not exist*");
```
for free. It's shorter, and tests more specific.

The `NMTST_EXPECT_LIBNM_CRITICAL` variant exists for completeness and consistency. If you already understand what `NMTST_EXPECT_NM_WARN` does, the libnm version should look familiar. What it provides, is that it encapsulates the `"libnm"` domain, so vim autosuggests the function with `NMTST_E<C-n>`, instead of `g_test_e<C-n> ("libnm", G_LOG_L<C-n>, `.

Reply:
> I mostly find the NMTST_EXPECT_NM_* macros useful, because they also check for the right logging level, so we get
> 
> -    g_test_expect_message ("NetworkManager", G_LOG_LEVEL_MESSAGE,
> -                           "*<warn> * keyfile: 802-1x.client-cert: certificate or key file '/CASA/dcbw/Desktop/certinfra/client.pem' does not exist*");
> +    NMTST_EXPECT_NM_WARN ("keyfile: 802-1x.client-cert: certificate or key file '/CASA/dcbw/Desktop/certinfra/client.pem' does not exist*");
> 
> for free. It's shorter, and tests more specific.

Okay, it's fine with me.